src/js/Controllers/UIController.js

```diff
@@ -786,31 +795,31 @@ class UIController {
     onKeepContext(alert, isSubtle) {
         clearTimeout(this.timers[alert.msgID])
         this.onButtonPress(alert.appID, alert.buttonID, alert.buttonName)
-        var timeout = alert.duration ? alert.duration : 10000
+        const timeout = store.getState().system.resetPeriod;
```

Comment:
```suggestion
        const timeout = alert.duration ? alert.duration : 10000;
```

Reply:
@iCollin please find the changes [here](https://github.com/LuxoftSDL/generic_hmi/pull/39/files).